server/src/main/java/org/apache/druid/client/cache/ForegroundCachePopulator.java

```diff
@@ -36,7 +36,6 @@
 {
```

Comment:
Please provide [concurrent access documentation](https://github.com/code-review-checklists/java-concurrency#justify-document) for this class (or for `CachePopulator` in general, and make `ForegroundCachePopulator`'s Javadoc just refer to its superclass's Javadoc), and/or describe the [concurrent control or data flow](https://github.com/code-review-checklists/java-concurrency#threading-flow-model).

In particular, this passage from the PR description:
> since CachePopulator is shared by all processing threads but the lock covers bytes and json generator that are tied to a specific segment which should only be happening in a single thread.

is not obvious, but it's not reflected anywhere in the code itself.

Reply:
I think there isn't so much concurrent access to document anymore since all concurrency primitives have been removed from `ForegroundCachePopulator`, which I think previously qualified for the [excessive thread safety](https://github.com/code-review-checklists/java-concurrency#excessive-thread-safety) section. I have tried to update the javadocs of `CachePopulator` to describe it's usage and how the sequence is consumed, so hopefully at least I've left it better than I found it.